src/lmic/lmic.c

```diff
@@ -880,14 +880,26 @@ static void initJoinLoop (void) {
 static ostime_t nextJoinState (void) {
     // Try the following:
     //   SF7/8/9/10  on a random channel 0..63
-    //   SF8C        on a random channel 64..71
+    //			(honoring enable mask)
+    //   SF8C     on a random 500 kHz channel 64..71
+    //			(always determined by
+    //			 previously selected 
+    //			 125 kHz channel)
     //
     u1_t failed = 0;
     if( LMIC.datarate != DR_SF8C ) {
+        // assume that 500 kHz equiv of last 125 kHz channel 
+	// is also enabled, and use it next.
         LMIC.txChnl = 64+(LMIC.txChnl&7);
```

Comment:
@terrillmoore Shouldn't this line be ```LMIC.txChnl = 64+(LMIC.txChnl >> 3);```?

Maybe we should define something like this:
```
#define SubBandOfChannel(c) (c<64 ? (c>>3) : (c & 7))
```